src/dblib/dblib.c

```diff
@@ -1258,13 +1241,115 @@ tdsdbopen(LOGINREC * login, const char *server, int msdblib)
 		return NULL;
 	}
 
-	connection = tds_read_config_info(dbproc->tds_socket, login->tds_login, g_dblib_ctx.tds_ctx->locale);
-	if (!connection) {
-		dbclose(dbproc);
-		return NULL;
+	// Open connection direct while the server is empty
+	if (NULL == server) {
+		/*
+		* Sybase supports the DSQUERY environment variable and falls back to "SYBASE" if server is NULL. 
+		* Microsoft uses a NULL or "" server to indicate a local server.  
+		* FIXME: support local server for win32.  
+		*/
```

Comment:
this is the expected behaviour, you are changing it

Reply:
I'm enhancing it, while the server is null and env is empty, then it will fail. 

But I have passed all connection args via LOGIN